hedyorg/hedy
# 💻 Add Ability to Rename Class

Fixes #5965 

Added functionality to rename a class, going off the instructions in the issue comments with direct editing from change on the input element.

I edited the exisiting rename_class function to directly send a request instead of opening a modal, as per the comment in the issue, but I believe it may be good to add a confirmation modal as well, let me know

Also, I wasn't sure what to do with the rename_class in the teachers.ts file, which does not seemed to be used at all, so I left it unchanged

**How to test**

Follow these steps to verify this PR works as intended:

* Log in as a teacher
* create a class
* in the customize class page, edit the input box, and click outside the box or press enter to confirm
* verify the name has been changed by going back 

**Dev Testing**
https://github.com/user-attachments/assets/422caab4-89d8-4d75-8a08-03899de1d5bb

**Checklist**
